### PR TITLE
Make the RegisterServiceWithConsul function configurable by options

### PR DESCRIPTION
**Pull request by aligator**
I kept it compatible with the old version, added optional options. So no major version update is needed.

### CODE REVIEW

File: consul.go [hunk 1]
```diff
@@ -43,32 +111,15 @@ func RegisterServiceWithConsul(serviceName string) {
 	registration.Name = serviceName
 	address := Hostname()
 	registration.Address = address
-	port, err := strconv.Atoi(Port()[1:len(Port())])
+	port, err := strconv.Atoi(Port(cfg.defaultPort)[1:len(Port(cfg.defaultPort))])
```
Comment (dominikbraun) [inline consul.go:114]: Please explain this line in a brief comment, ideally with an example.

File: consul.go [hunk 2]
```diff
@@ -27,13 +27,81 @@ func connect() *api.Client {
...
+}
+
+// WithHTTPHealthCheck enables a health check using a simple small webserver
 // which gets automatically started.
-// To configure the Port you can pass
-// PRODUCT_SERVICE_PORT and PRODUCT_HEALTH_PORT as environment variable.
-// The default ports are 8100 and 8101.
-func RegisterServiceWithConsul(serviceName string) {
+// The default port setting can always be overwritten by an environment variable named PRODUCT_HEALTH_PORT.
+func WithHTTPHealthCheck(defaultPort int) Option {
```
Comment (dominikbraun) [inline consul.go:60]: You may note in the GoDoc that this function starts a webserver for the sake of clarity.

**Review by dominikbraun — CHANGES_REQUESTED** [review]

File: consul.go [hunk 3]
```diff
@@ -27,13 +27,81 @@ func connect() *api.Client {
...
+	return WithRegistrationModifier(func(registration *api.AgentServiceRegistration) {
+		// setup simple health detection using a small webserver
+		registration.Check = new(api.AgentServiceCheck)
+		healthPortNr, err := strconv.Atoi(HealthPort(defaultPort)[1:len(HealthPort(defaultPort))])
+		if err != nil {
+			log.Fatalf("wrong healt Port format %v", err)
```
Comment (HilmerChristophAkaHuima) [inline consul.go:66]:
```suggestion
			log.Fatalf("wrong health port format %v", err)
```

File: consul.go (hunk 2 above)
Comment (dominikbraun) [reply consul.go:60]: Nevermind!

File: consul.go [hunk 4]
```diff
@@ -27,13 +27,81 @@ func connect() *api.Client {
...
+			log.Fatalf("healthcheck webserver failed %v", err)
+		}()
+	})
+}
+
+// RegisterServiceWithConsul registers a new service to consul.
+//
+// Deprecated: To use a simple HTTP health service, use WithHTTPHealthCheck().
+// Currently it uses WithHTTPHealthCheck automatically if no other RegistrationModifier is used.
+// From v1.0.0 on RegisterServiceWithConsul will not do this automatically anymore.
```
Comment (aligator) [inline consul.go:89]:
```suggestion
// From v1.0.0 on RegisterServiceWithConsul will not do this automatically anymore.
// To disable the health check completely (in < v1.0.0) just pass an empty registration modifier: WithRegistrationModifier(func(_){})
```

File: consul.go [hunk 5]
```diff
@@ -27,13 +27,81 @@ func connect() *api.Client {
...
+		go func() {
+			err := http.ListenAndServe(HealthPort(defaultPort), nil)
+			log.Fatalf("healthcheck webserver failed %v", err)
+		}()
+	})
+}
+
+// RegisterServiceWithConsul registers a new service to consul.
+//
+// Deprecated: To use a simple HTTP health service, use WithHTTPHealthCheck().
```
Comment (aligator) [inline consul.go:87]:
```suggestion
// Deprecation note: To use a simple HTTP health service, use WithHTTPHealthCheck().
```

File: consul.go [hunk 6]
```diff
@@ -43,32 +111,15 @@ func RegisterServiceWithConsul(serviceName string) {
 	registration.Name = serviceName
 	address := Hostname()
 	registration.Address = address
-	port, err := strconv.Atoi(Port()[1:len(Port())])
+	port, err := strconv.Atoi(Port(cfg.defaultPort)[1:len(Port(cfg.defaultPort))])
 	if err != nil {
 		log.Fatalf("wrong Port format %v", err)
```
Comment (HilmerChristophAkaHuima) [inline consul.go:116]: Port => port

File: consul.go [hunk 7]
```diff
@@ -97,18 +150,18 @@ func GetServicesWithConsul(serviceName string) []*api.ServiceEntry {
 	return services
 }
 
-func Port() string {
+func Port(defaultPort int) string {
```
Comment (HilmerChristophAkaHuima) [inline consul.go:153]: Maybe without ":"

File: consul.go [hunk 8]
```diff
@@ -97,18 +150,18 @@ func GetServicesWithConsul(serviceName string) []*api.ServiceEntry {
...
 	p := os.Getenv("PRODUCT_SERVICE_PORT")
 	if len(strings.TrimSpace(p)) == 0 {
-		return ":8100"
+		return ":" + strconv.Itoa(defaultPort)
 	}
 	return fmt.Sprintf(":%s", p)
 }
 
-func HealthPort() string {
+func HealthPort(defaultPort int) string {
```
Comment (HilmerChristophAkaHuima) [inline consul.go:161]: Maybe without ":"

**Review by HilmerChristophAkaHuima — CHANGES_REQUESTED** [review]
Some spelling errors and comments on the Port func

File: consul.go (hunk 1 above)
Comment (aligator) [reply consul.go]: changed the Port function completely

File: consul.go (hunk 7 above)
Comment (aligator) [reply consul.go:171]: replaced function completely

File: consul.go (hunk 6 above)
Comment (aligator) [reply consul.go]: this log doesn't exist anymore

**Review by HilmerChristophAkaHuima — APPROVED** [review]
All good :smile:

**Review by dominikbraun — DISMISSED** [review]
LGTM

**Review by dominikbraun — DISMISSED** [review]
LGTM

**Review by HilmerChristophAkaHuima — APPROVED** [review]
:metal:

**Review by muhmuhhum — COMMENTED** [review]
basst

**Review by muhmuhhum — APPROVED** [review]

**Pull request merged**